### DISCUSSION TITLE
UsdGeomXformCommonAPI too restrictive?

### DISCUSSION THREAD

**Discussion by pierre-ed** [Autodesk/maya-usd#995] [Q&A]
Hello everyone!

While trying to edit Xforms from various USD stages imported as proxyShapes in Maya, I encountered an issue, namely that `maya_usd` currently only interactively manipulates Xforms that are compliant to the `UsdGeomXformCommonAPI` schema.  

Unfortunately, most other DCC's produce incompatible Xforms, either expressed only as matrices or with arbitrarily-named xform ops. The list is actually rather long: Solaris, Blender, mtoa among others, and even the `al` exporter from `maya_usd` itself (which produces `translate:rotatePivot` and `translate:scalePivot` ops).

Has anyone encountered this problem before?
I would like to know what you guys think about this issue, which, paradoxically, kind of predates the interchangeability objective of USD. 
Is this requirement on Xform description is likely to get lifted by `maya_usd` itself in a near future? 
And in order to circumvent my issue, would you advise me to preprocess my USD data before importing it into Maya or instead change the `UsdMayaEditUtil::ApplyEditsToProxy()` implementation directly?  

As always, thank you all for your help.
Cheers,
-- Pierre

**Comment by kxl-adsk** [comment]
The timing couldn't be any better. We were are of this limitation and were working on. Last weekend we merged this work to the dev branch - https://github.com/Autodesk/maya-usd/pull/962 - but to fully test it, you need to wait for the next Maya Preview Release (i.e. PR 121).

Give it a try and let us know how it works for you! It's the perfect time to provide feedback.
